diff --git a/src/http/ngx_http_upstream.h b/src/http/ngx_http_upstream.h
index bcdd64b..84e92e3 100644
--- a/src/http/ngx_http_upstream.h
+++ b/src/http/ngx_http_upstream.h
@@ -106,6 +106,14 @@ struct ngx_http_upstream_srv_conf_s {
 
     ngx_array_t                     *servers;  /* ngx_http_upstream_server_t */
 
+#if (NGX_HTTP_UPSTREAM_HASH) 
+    ngx_array_t                     *values;
+    ngx_array_t                     *lengths;
+    ngx_uint_t                       retries;
+    ngx_uint_t                       port_hi;
+    ngx_uint_t                       port_lo;
+#endif
+
     ngx_uint_t                       flags;
     ngx_str_t                        host;
     u_char                          *file_name;
